pkg-r/R/querychat-package.R

```diff
@@ -1,7 +1,61 @@
+#' querychat: Chat with Your Data Using Natural Language
+#'
+#' @description
+#' querychat provides an interactive chat interface for querying data using
+#' natural language. It translates your questions into SQL queries, executes
+#' them against your data, and displays the results. The package works with
+#' both data frames and database connections.
+#'
+#' @section Quick Start:
+#' The easiest way to get started is with the [QueryChat] R6 class:
+#'
+#' ```r
+#' library(querychat)
+#'
+#' # Create a QueryChat object
+#' qc <- QueryChat$new(mtcars, "mtcars")
+#'
+#' # Option 1: Run a complete app with sensible defaults
+#' qc$app()
+#'
+#' # Option 2: Build a custom Shiny app
+#' ui <- page_sidebar(
+#'   qc$sidebar(),
+#'   dataTableOutput("data")
+#' )
+#'
+#' server <- function(input, output, session) {
+#'   qc$server()
+#'   output$data <- renderDataTable(qc$df())
+#' }
+#'
+#' shinyApp(ui, server)
+#' ```
+#'
+#' @section Key Features:
+#' - **Natural language queries**: Ask questions in plain English
+#' - **SQL transparency**: See the generated SQL queries
+#' - **Multiple data sources**: Works with data frames and database connections
+#' - **Customizable**: Add data descriptions, extra instructions, and custom greetings
+#' - **LLM agnostic**: Works with OpenAI, Anthropic, Google, and other providers via ellmer
+#'
+#' @section Main Components:
+#' - [QueryChat]: The main R6 class for creating chat interfaces
+#' - [create_data_source()]: (Advanced) Create custom data source objects
+#'
+#' @section Examples:
+#' See the package examples directory for complete working apps:
+#' - `01-hello-app/`: Minimal example using `$app()`
+#' - `02-sidebar-app/`: Custom layout with additional UI elements
+#' - `sqlite/`: Database connection example
```

Comment:
I like these sections, nice addition to the package docs! It'd be worth mentioning how to use `shiny::runExample()` to run these examples